File: src/ympa_csharp_functions_server/Functions/OutletsApi.cs
using System.IO;
using System.Net;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
using Microsoft.Extensions.Logging;
using Microsoft.OpenApi.Models;
using Newtonsoft.Json;
using Org.OpenAPITools.Models;

namespace ympa_csharp_functions_server.Functions
{ 
    public partial class OutletsApi
    { 
        [FunctionName("OutletsApi_CreateOutlet")]
        public async Task<ActionResult<CreateOutletResponse>> _CreateOutlet([HttpTrigger(AuthorizationLevel.Anonymous, "Post", Route = "campaigns/{campaignId}/outlets")]HttpRequest req, ExecutionContext context, long campaignId)
        {
            var method = this.GetType().GetMethod("CreateOutlet");
            return method != null
                ? (await ((Task<CreateOutletResponse>)method.Invoke(this, new object[] { req, context, campaignId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("OutletsApi_DeleteOutlet")]
        public async Task<ActionResult<EmptyApiResponse>> _DeleteOutlet([HttpTrigger(AuthorizationLevel.Anonymous, "Delete", Route = "campaigns/{campaignId}/outlets/{outletId}")]HttpRequest req, ExecutionContext context, long campaignId, long outletId)
        {
            var method = this.GetType().GetMethod("DeleteOutlet");
            return method != null
                ? (await ((Task<EmptyApiResponse>)method.Invoke(this, new object[] { req, context, campaignId, outletId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("OutletsApi_GetOutlet")]
        public async Task<ActionResult<GetOutletResponse>> _GetOutlet([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/{campaignId}/outlets/{outletId}")]HttpRequest req, ExecutionContext context, long campaignId, long outletId)
        {
            var method = this.GetType().GetMethod("GetOutlet");
            return method != null
                ? (await ((Task<GetOutletResponse>)method.Invoke(this, new object[] { req, context, campaignId, outletId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("OutletsApi_GetOutlets")]
        public async Task<ActionResult<GetOutletsResponse>> _GetOutlets([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/{campaignId}/outlets")]HttpRequest req, ExecutionContext context, long campaignId)
        {
            var method = this.GetType().GetMethod("GetOutlets");
            return method != null
                ? (await ((Task<GetOutletsResponse>)method.Invoke(this, new object[] { req, context, campaignId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("OutletsApi_UpdateOutlet")]
        public async Task<ActionResult<EmptyApiResponse>> _UpdateOutlet([HttpTrigger(AuthorizationLevel.Anonymous, "Put", Route = "campaigns/{campaignId}/outlets/{outletId}")]HttpRequest req, ExecutionContext context, long campaignId, long outletId)
        {
            var method = this.GetType().GetMethod("UpdateOutlet");
            return method != null
                ? (await ((Task<EmptyApiResponse>)method.Invoke(this, new object[] { req, context, campaignId, outletId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }
    }
}

File: src/main/java/org/openapitools/server/api/verticle/OutletsApi.java
package org.openapitools.server.api.verticle;

import org.openapitools.server.api.model.ApiClientDataErrorResponse;
import org.openapitools.server.api.model.ApiForbiddenErrorResponse;
import org.openapitools.server.api.model.ApiLimitErrorResponse;
import org.openapitools.server.api.model.ApiNotFoundErrorResponse;
import org.openapitools.server.api.model.ApiServerErrorResponse;
import org.openapitools.server.api.model.ApiUnauthorizedErrorResponse;
import org.openapitools.server.api.model.ChangeOutletRequest;
import org.openapitools.server.api.model.CreateOutletResponse;
import org.openapitools.server.api.model.EmptyApiResponse;
import org.openapitools.server.api.model.GetOutletResponse;
import org.openapitools.server.api.model.GetOutletsResponse;
import org.openapitools.server.api.MainApiException;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

import java.util.List;
import java.util.Map;

public interface OutletsApi  {
    //createOutlet
    void createOutlet(Long campaignId, ChangeOutletRequest changeOutletRequest, Handler<AsyncResult<CreateOutletResponse>> handler);

    //deleteOutlet
    void deleteOutlet(Long campaignId, Long outletId, Handler<AsyncResult<EmptyApiResponse>> handler);

    //getOutlet
    void getOutlet(Long campaignId, Long outletId, Handler<AsyncResult<GetOutletResponse>> handler);

    //getOutlets
    void getOutlets(Long campaignId, String pageToken, Long regionId, String shopOutletCode, Long regionId2, Handler<AsyncResult<GetOutletsResponse>> handler);

    //updateOutlet
    void updateOutlet(Long campaignId, Long outletId, ChangeOutletRequest changeOutletRequest, Handler<AsyncResult<EmptyApiResponse>> handler);

}
